src/components/ConversationSettings/ConversationAvatarEditor.vue

```diff
@@ -0,0 +1,310 @@
+			const file = e.target.files[0]
+			if (!this.validMimeTypes.includes(file.type)) {
+				showError(t('spreed', 'Please select a valid png or jpg file'))
+				this.cancel()
+				return
+			}
+
+			const reader = new FileReader()
+			reader.onload = (e) => {
+				this.$refs.cropper.replace(e.target.result)
+				this.showCropper = true
+			}
+			reader.readAsDataURL(file)
+		},
+
+		async openFilePicker() {
+			const path = await picker.pick()
+			this.loading = true
+			try {
+				const { data } = await axios.post(generateUrl('/avatar'), { path })
+				if (data.status === 'success') {
+					this.loading = false
+				} else if (data.data === 'notsquare') {
+					const tempAvatar = generateUrl('/avatar/tmp') + '?requesttoken=' + encodeURIComponent(OC.requestToken) + '#' + Math.floor(Math.random() * 1000)
+					this.$refs.cropper.replace(tempAvatar)
+					this.showCropper = true
+				} else {
+					showError(data.data.message)
+					this.cancel()
+				}
+			} catch (e) {
+				showError(t('spreed', 'Error setting profile picture'))
+				this.cancel()
+			}
+		},
+
+		saveAvatar() {
+			this.showCropper = false
+			this.loading = true
+			const canvasData = this.$refs.cropper.getCroppedCanvas()
+			const scaleFactor = canvasData.width > 512 ? 512 / canvasData.width : 1
+			this.$refs.cropper.scale(scaleFactor, scaleFactor).getCroppedCanvas().toBlob(async (blob) => {
+				if (blob === null) {
+					showError(t('spreed', 'Error cropping profile picture'))
+					this.cancel()
+					return
+				}
+
+				const formData = new FormData()
+				formData.append('file', blob)
+
+				try {
+					await this.$store.dispatch('setConversationPictureAction', {
+						token: this.conversation.token,
+						file: formData,
+					})
+					this.loading = false
+				} catch (e) {
+					showError(t('spreed', 'Error saving profile picture'))
+					this.loading = false
+				}
+			})
+		},
+
+		async removeAvatar() {
+			this.loading = true
+			try {
+				await this.$store.dispatch('deleteConversationPictureAction', {
+					token: this.conversation.token,
+				})
+				this.loading = false
+			} catch (e) {
+				showError(t('spreed', 'Error removing profile picture'))
+				this.loading = false
+			}
+		},
+
+		cancel() {
+			this.showCropper = false
+			this.loading = false
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+section {
+	grid-row: 1/3;
+}
+.avatar {
+	&__container {
+		margin: 0 auto;
+		display: flex;
+		flex-direction: column;
+		justify-content: center;
+		align-items: center;
+		gap: 16px 0;
+		width: 300px;
+
+		span {
+			color: var(--color-text-lighter);
+		}
+	}
+
+	&__preview {
+		display: flex;
+		justify-content: center;
+		align-items: center;
+		width: 180px;
+		height: 180px;
+	}
+
+	&__buttons {
+		display: flex;
+		gap: 0 10px;
+	}
+
+	&__cropper {
+		width: 300px;
+		height: 300px;
+		overflow: hidden;
+
+		&-buttons {
+			width: 100%;
+			display: flex;
+			justify-content: space-between;
+		}
+
+		&::v-deep .cropper-view-box {
```

Comment:
```suggestion
		& :deep(.cropper-view-box) {
```

`v-deep` is deprecated